mail_template_default/wizard/mail_compose_message.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 Solvti sp. z o.o. (https://solvti.pl)
+
+from odoo import api, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class MailComposer(models.TransientModel):
+    _inherit = "mail.compose.message"
+
+    @api.model
+    def default_get(self, fields):
+        """Mail Template Rules are sorted base on 4 fields in following hierarchy:
+        bool(context_flag) -> bool(company_id) -> bool(field_domain) -> sequence
+        The table show hypothetically order:
+
+        +---------+---------+--------+-----+
+        | context | company | domain | seq |
+        +---------+---------+--------+-----+
+        |    x    |    x    |   x    |  1  |
+        |    x    |    x    |   x    |  2  |
+        |    x    |         |        |  2  |
+        |         |    x    |   x    |  1  |
+        |         |    x    |        |  1  |
+        |         |         |   x    |  2  |
+        |         |         |   x    |  3  |
+        |         |         |        |  1  |
+        """
+
+        res = super(MailComposer, self).default_get(fields)
+
+        res_model = self._context.get("active_model") or self._context.get(
+            "default_model"
+        )
+        res_id = self._context.get("active_id") or self._context.get("default_res_id")
+        force_mt = self._context.get("force_mail_template")
+        domain = []
+        if not res_model or not res_id or force_mt is False:
```

Comment:
If you want to use is, it better to use None, simple not or check if key is in context. So, some more clear check.